Core/GDCore/Project/Layout.h

```diff
@@ -443,15 +443,33 @@ gd::String GD_CORE_API GetTypeOfObject(const ObjectsContainer& game,
                                        gd::String objectName,
                                        bool searchInGroups = true);
 /**
- * \brief Check if an object or all object of a group has a behavior.
+ * \brief Check if an object or all objects of a group has a behavior.
  */
 bool GD_CORE_API HasBehaviorInObjectOrGroup(const gd::ObjectsContainer &project,
                                             const gd::ObjectsContainer &layout,
                                             const gd::String &objectOrGroupName,
                                             const gd::String &behaviorName,
                                             bool searchInGroups = true);
 /**
- * \brief Get the type of a behavior if an object or some object of a group has it.
+ * \brief Get the names of behavior of a given type if an object or all objects of a group has it.
+ */
+std::vector<gd::String> GD_CORE_API GetBehaviorNamesInObjectOrGroup(

```

Comment:
What is this used for?

Reply:
It allows to check if an object or a group have default behaviors that match following behavior parameters.
This is used to:
- Filter object parameters autocompletion
- Show error on object parameters that are filled with an existing object that doesn't match required default behaviors.